tmt/steps/provision/mrack.py

```diff
@@ -680,6 +680,8 @@ class ProvisionBeaker(tmt.steps.provision.ProvisionPlugin[ProvisionBeakerData]):
     _data_class = ProvisionBeakerData
     _guest_class = GuestBeaker
 
+    # _thread_safe = True
```

Comment:
Uncomment it, use _thread_safe = True, beaker parallel provisioning works for me. Wondering why you comment it?

[client-1]         guest: has been requested
[client-1]         job id: 8784929
[client-1]         wait: waiting for condition 'get_new_state' with timeout 1:00:00, deadline in 3600.0 seconds, checking every 60.00 seconds
[server-1]         guest: has been requested
[server-1]         job id: 8784930
[server-1]         wait: waiting for condition 'get_new_state' with timeout 1:00:00, deadline in 3600.0 seconds, checking every 60.00 seconds
[client-2]         guest: has been requested
[client-2]         job id: 8784931
...
summary: 3 guests provisioned
            client-2
            client-1
            server-1

Reply:
Because we are not sure whether the library tmt uses for Beaker provisioning, mrack, is thread-safe. We did run into issues with the `virtual` plugin, where the problem was eventually solved by adding an extra lock to provide serialization, but there was no time invested into doing the same for the Beaker plugin. If it's required, that is.

Filed https://github.com/teemtee/tmt/issues/2607 to track this.